agent/http_test.go

```diff
@@ -462,15 +463,29 @@ func TestParseConsistency(t *testing.T) {
 	if !b.RequireConsistent {
 		t.Fatalf("Bad: %v", b)
 	}
+
+	// check using stale by default for http
+	var query structs.QueryOptions
+	a.srv.agent.config.HTTPAPIStaleByDefault = true
+
+	req, _ = http.NewRequest("GET", "/v1/catalog/nodes", nil)
+	if d := a.srv.parseConsistency(resp, req, &query); d {
+		t.Fatalf("unexpected done")
+	}
+
+	if !query.AllowStale {
+		t.Fatalf("Bad: %v", query)
```

Comment:
Make the message be a bit more clear, maybe something like "Expected allowStale to be true but was false", rather than printing the whole query string.